cmd/harmony/main.go

```diff
@@ -140,6 +139,11 @@ func initSetup() {
 
 func createGlobalConfig() *nodeconfig.ConfigType {
 	nodeConfig := nodeconfig.GetGlobalConfig()
+	nodeConfig.StakingPriKey = node.LoadStakingKeyFromFile(*stakingKeyFile, *accountIndex)
+
+	// Initialize leveldb if dbSupported.
+	nodeConfig.MainDB, _ = InitLDBDatabase(*ip, *port, *freshDB, false)
```

Comment:
Let's not ignore errors.

Reply:
(I know you are just refactoring existing code, but let's DTRT when we touch something and leave it cleaner than it was. 😄)

Reply:
Also, technically `MainDB` is a live object that holds reference to system resource (on-disk file) and not simply a data-only configuration parameter.  (`ip`, `port`, and `freshDB` are.)  I'd suggest inlining this and eliminating it from config object.

Reply:
Fixing the error.

As for the MainDB i agree it maybe not the best setup. I may need to rework on this but the goal of this PR is to generate consensus and node struct based on single node config. I'm even still not happy with the node config yet but it's the starter effort.